opmd_viewer/addons/pic/lpa_diagnostics.py

```diff
@@ -17,4 +20,668 @@ def __init__(self, path_to_dir):
+
+        species : string
+            Particle species to use for calculations
+
+        select : dict, optional
+            Either None or a dictionary of rules
+            to select the particles, of the form
+            'x' : [-4., 10.]   (Particles having x between -4 and 10 microns)
+            'z' : [0, 100] (Particles having x between 0 and 100 microns)
+
+        Returns
+        -------
+        A tuple of floats with:
+        - mean weighted gamma
+        - weighted standard deviation of gamma
+        """
+        # Find the output that corresponds to the requested time/iteration
+        # (Modifies self.current_i and self.current_t)
+        self._find_output( t, iteration )
+        # Get particle data
+        ux, uy, uz, w = self.get_particle(
+                         var_list=['ux', 'uy', 'uz', 'w'],
+                         species=species, t=t, iteration=iteration )
+        # Calculate Lorentz factor for all particles
+        gamma = np.sqrt(1 + ux ** 2 + uy ** 2 + uz ** 2)
+        # Calculate weighted mean and average
+        try:
+            # Calculate mean_gamma for selected particles
+            mean_gamma = np.sqrt(np.average(gamma ** 2, weights=w))
+        except ZeroDivisionError:
+            # If selection is empty or all particles have weight zero,
+            # return NaN
+            mean_gamma = np.nan
+        std_gamma = wstd(gamma, w)
+        # Return the result
+        return( mean_gamma, std_gamma )
+
+    def get_charge( self, t=None, iteration=None, species=None, select=None ):
+        """
+        Calculate the charge of the selcted particles.
+
+        Parameters
+        ----------
+        t : float (in seconds), optional
+            Time at which to obtain the data (if this does not correspond to
+            an available file, the last file before `t` will be used)
+            Either `t` or `iteration` should be given by the user.
+
+        iteration : int
+            The iteration at which to obtain the data
+            Either `t` or `iteration` should be given by the user.
+
+        species : string
+            Particle species to use for calculations
+
+        select : dict, optional
+            Either None or a dictionary of rules
+            to select the particles, of the form
+            'x' : [-4., 10.]   (Particles having x between -4 and 10 microns)
+            'z' : [0, 100] (Particles having x between 0 and 100 microns)
+
+        Returns
+        -------
+        A float with the electric charge of the selected particles in Coulomb
+        """
+        # Find the output that corresponds to the requested time/iteration
+        # (Modifies self.current_i and self.current_t)
+        self._find_output( t, iteration )
+        # Get particle data
+        w, q = self.get_particle( var_list=['w', 'charge'], species=species,
+                                 t=t, iteration=iteration )
+        # Calculate charge
+        charge = np.sum(w) * q
```

Comment:
One thing to keep in mind is that `q` could in principle be an array (with size = the number of macroparticles) with different values of the charge (for instance, a bunch of ions that have been ionized to various degrees) - as openPMD allows for that kind of thing. For this reason, I would replace
```python
charge = np.sum(w) * q
```
by
```python
charge = np.sum(w * q)
```
That way, the final result is bound to be a scalar, not an array. 
Is that okay with you ?